app/models/playlist.rb

```diff
@@ -18,15 +18,15 @@ class Playlist < ApplicationRecord
   enum access: { shared: 0, personal: 1, secret: 2 }
   enum playlist_type: { album: 0, artist: 1, track: 2 }
 
-  scope :with_item_ids, -> { left_joins(:items).select('playlists.*', 'array_agg(playlist_items.item_id ORDER BY playlist_items.order ASC) as item_ids').group(:id) }
+  scope :with_item_ids, -> { left_joins(:items).select('playlists.*', 'array_remove(array_agg(playlist_items.item_id ORDER BY playlist_items.order ASC), NULL) as item_ids').group(:id) }
```

Comment:
Can you add a comment in the code on why the `array_remove(..., NULL)` is necessary? It isn't immediately obvious to me now, so it definitely won't be in the future.